packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,249 @@
+          ' ' +
+          field.name
+      );
+    } else {
+      if (field.referenceTo.length > 1) {
+        decls.push('SObject' + ' ' + field.relationshipName);
+      } else {
+        decls.push(field.referenceTo + ' ' + field.relationshipName);
+      }
+      // field.type will be "reference", but the actual type is an Id for Apex
+      decls.push('Id ' + field.name);
+    }
+    return decls;
+  }
+
+  private static fieldName(decl: string) {
+    return decl.substr(decl.indexOf(' ') + 1);
+  }
+
+  // VisibleForTesting
+  public async generateFauxClass(
+    folderPath: string,
     sobject: any
-  ): Promise<void> {
-    const folderPath = path.join(projectPath, '.sfdx', 'sobjects');
+  ): Promise<string> {
     if (!fs.existsSync(folderPath)) {
-      //fs.mkdirSync(folderPath);
+      fs.mkdirSync(folderPath);
     }
-    //const isCustom = sobject.custom;
+    const fauxClassPath = path.join(folderPath, sobject.name + '.cls');
+
+    fs.writeFileSync(fauxClassPath, this.generateFauxClassText(sobject));
+
+    return fauxClassPath;
+  }
+
+  private generateFauxClassDecls(sobject: SObject): string[] {
+    const declarations: string[] = [];
+    if (sobject.fields) {
+      for (const field of sobject.fields) {
+        const decls: string[] = this.generateField(field);
+        if (decls && decls.length > 0) {
+          for (const decl of decls) {
+            declarations.push(decl);
+          }
+        }
+      }
+    }
+
+    if (sobject.childRelationships) {
+      for (const rel of sobject.childRelationships) {
+        if (rel.relationshipName) {
+          const decl: string = this.generateChildRelationship(rel);
+          if (decl) {
+            declarations.push(decl);
+          }
+        }
+      }
+      for (const rel of sobject.childRelationships) {
+        // handle the odd childRelationships last (without relationshipName)
+        if (!rel.relationshipName) {
+          const decl: string = this.generateChildRelationship(rel);
+          if (decl) {
+            declarations.push(decl);
+          }
+        }
+      }
+    }
+    return declarations;
+  }
+
+  private generateFauxClassTextFromDecls(
+    className: string,
+    declarations: string[]
+  ): string {
+    // sort, but filter out duplicates
+    // which can happen due to childRelationships w/o a relationshipName
+    declarations.sort(function nameCompare(
+      first: string,
+      second: string
+    ): number {
+      return FauxClassGenerator.fieldName(first) >
+      FauxClassGenerator.fieldName(second)
+        ? 1
+        : -1;
+    });
+
+    declarations = declarations.filter(function checkDups(
+      value: string,
+      index: number,
+      array: string[]
+    ) {
+      return (
+        !index ||
+        FauxClassGenerator.fieldName(value) !==
+          FauxClassGenerator.fieldName(array[index - 1])
+      );
+    });
+
+    const indentAndModifier = '    global ';
+    const generatedClass: string =
```

Comment:
I don't think we should hardcode '\n'. This is unlikely to work on Windows.

Reply:
Good point - I did see it used in other places (localization.ts and a couple of places in linter)...    I'll try to come up with a better way.